packages/caver-wallet/src/keyring/keyring.js

```diff
@@ -0,0 +1,845 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const _ = require('lodash')
+const scrypt = require('scrypt-shim')
+const uuid = require('uuid')
+const cryp = typeof global === 'undefined' ? require('crypto-browserify') : require('crypto')
+const AccountLib = require('eth-lib/lib/account')
+
+const utils = require('../../../caver-utils')
+const PrivateKey = require('./privateKey')
+const { KEY_ROLE } = require('./keyringHelper')
+const Account = require('../../../caver-account')
+
+/**
+ * representing a Keyring which includes `address` and `private keys` by roles.
+ * @class
+ */
+class Keyring {
+    /**
+     * generates a private key string
+     *
+     * `caver.wallet.keyring.generatePrivateKey()`
+     *
+     * @param {string} entropy A random string to increase entropy.
+     * @return {string}
+     */
+    static generatePrivateKey(entropy) {
```

Comment:
Is this function necessary? I think we can implement this using `caver.wallet.keyring.generate().key()`?

Reply:
This function is not necessary, but if someone want multiple private key strings, I think it will be more convenient to use.

Reply:
Hmm.. Is any other way to get random private key except this?

Reply:
There is static method named 'generate' in PrivateKey class. But use cannot access to this class.

Reply:
What about making that function public? Actually generate private key is out of scope of this keyring class.

Reply:
To do this, the user must have direct access to the PrivateKey class.
"caver.wallet.keyring.privateKey.generate()" like this. is this fine?

Reply:
@jimni1222 It seems too deep to call the function, but developers can import the privateKey module like require("....../privatekey"), right?

Reply:
In my opinion, it is rare to use the internal implementation directly by loading the module with require.
For example, const PrivateKey = require ('../../../ node_modules/caver-js/packages/caver-wallet/src/keyring/privateKey')

Reply:
@jimni1222 okay. then what about this?
`privateKey = cave.wallet.keyring.privateKey`
Is this possible?

Reply:
If the user can access to the PrivateKey class with caver.wallet.keyring.privateKey, the above usability is possible.

'caver.wallet.keyring.privateKey.generate()' and
'const PrivateKey = caver.wallet.keyring.privateKey; PrivateKey.xxx' are implemented identically and will be used differently depending on the style of the developer.

Do you think it would be better to modify the implementation to allow users to access the PrivateKey class like this?

Reply:
@jimni1222 

1. I don't think the usability is good even if we can use `caver.wallet.keyring.generatePrivateKey()`. In terms of usability, it would be better to use `caver.wallet.generatePrivateKey()`.
2. It is weird that keyring provides a function `generatePrivateKey()` which is not very related to `keyring`.

Reply:
It seems good to provide caver.wallet.generatePrivateKey() in the caver.wallet package.
I will update like that. :)

Reply:
I added generatePrivateKey function in keyringContainer. 
So that will be uploaded in next PR !